src/Presentation/Nop.Web.Framework/Infrastructure/Extensions/ApplicationBuilderExtensions.cs

```diff
@@ -89,6 +90,21 @@ public static async Task StartEngineAsync(this IApplicationBuilder _)
             //clear payment info requests
             var genericAttributeService = engine.Resolve<IGenericAttributeService>();
             await genericAttributeService.DeleteAttributesAsync<Customer>(NopCustomerDefaults.ProcessPaymentRequestAttribute);
+
+            // Publish AppStartedEvent in background to avoid delaying HTTP request handling
+            var eventPublisher = engine.Resolve<IEventPublisher>();
+            _ = Task.Run(async () =>
+            {
+                try
+                {
+                    await eventPublisher.PublishAsync(new AppStartedEvent());
+                }
+                catch (Exception ex)
+                {
+                    var logger = engine.Resolve<ILogger>();
```

Comment:
I think simply logging problems with processing the application start event at this level is also bad, for the same reason as processing the event in parallel